docs/developer-guide/Tests/e2e-quickstart.md

```diff
@@ -0,0 +1,138 @@
+
+Before getting started, ensure the following tools are installed in your environment:
+
+- **Go**: For running the test framework.
+- **Docker**: For containerizing applications.
+- **kubectl**: For managing Kubernetes clusters.
+
+## E2E Test Environment
+
+Kmesh E2E testing requires a two-node KinD cluster:
+
+- **Control Node**: Manages the cluster.
+- **Worker Node**: Runs the test services.
+
+At the start of the test, two services will be deployed:
+
+1. **service-with-waypoint-at-service-granularity**: A service with a Waypoint.
+2. **enrolled-to-kmesh**: A service without a Waypoint.
+
+Both services use Echo Pods, which are used to test different scenarios.
+
+## Writing E2E Tests
+
+Here is a simple E2E test function template:
+
+```go
+func TestEchoCall(t *testing.T) {
+    framework.NewTest(t).Run(func(t framework.TestContext) {
+        t.NewSubTest("Echo Call Test").Run(func(t framework.TestContext) {
+            // Retrieve test services
+            src := apps.ServiceWithWaypointAtServiceGranularity[0]
+            dst := apps.EnrolledToKmesh
+
+            // Define test cases
+            cases := []struct {
+                name string
+                checker echo.Checker
+            }{
+                {
+                    name: "basic call",
+                    checker: echo.And(echo.ExpectOK(), echo.ExpectBodyContains("Hello")),
+                },
+            }
+
+            // Execute test cases
+            for _, c := range cases {
+                t.NewSubTest(c.name).Run(func(t framework.TestContext) {
+                    src.CallOrFail(t, echo.CallOptions{
+                        Target:   dst[0],
+                        PortName: "http",
+                        Checker:  c.checker,
+                    })
+                })
+            }
+        })
+    })
+}
+```
+
+### Resource Cleanup
+
+Use the `t.Cleanup` method to ensure test resources are cleaned up after the test completes. For example:
+
+```go
+t.Cleanup(func() {
+    // Clean up resources
+})
+```
+
+### Deploying Policies
+
+Use the `t.ConfigIstio` method to deploy policies required for the test. For example:
+
+```go
+t.ConfigIstio().YAML("test-namespace", `
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-all
+spec:
+  rules:
+  - {}
+`).ApplyOrFail(t)
+```
+
+### Using echo.Checker
+
+`echo.Checker` is used to verify whether a test case passes. For example:
+
+```go
+// Example: Using echo.Checker to validate HTTP response
+src.CallOrFail(t, echo.CallOptions{
+    Target:   dst[0],
+    PortName: "http",
+    Checker: echo.And(
+        echo.ExpectOK(),                      // Expect the HTTP call to succeed
+        echo.ExpectBodyContains("Hello"),    // Expect the response body to contain "Hello"
+        echo.ExpectHeaders(map[string]string{
+            "Content-Type": "text/plain",    // Expect the Content-Type header to be "text/plain"
+        }),
+    ),
+})
+```
+
+## Running Tests
```

Comment:
This section has been described in https://kmesh.net/docs/developer-guide/Tests/e2e-test. Could reference it and expand it.